packages/graphql/tests/integration/directives/fulltext.int.test.ts

```diff
@@ -647,4 +647,99 @@ describe("@fulltext directive", () => {
             `@fulltext index '${indexName}' on Node '${type.name}' already exists, but is missing field 'description'`
         );
     });
+
+    test("should create index for ID field if it doesn't exist and then query using the index", async () => {
+        // Skip if multi-db not supported
+        if (!MULTIDB_SUPPORT) {
+            console.log("MULTIDB_SUPPORT NOT AVAILABLE - SKIPPING");
+            return;
+        }
+
+        const id = generate({ readable: true, charset: "alphabetic" });
+        const indexName = generate({ readable: true, charset: "alphabetic" });
+        const type = generateUniqueType("Movie");
+
+        const typeDefs = gql`
+            type ${type.name} @fulltext(indexes: [{ name: "${indexName}", fields: ["id"] }]) {
+                id: ID!
+            }
+        `;
+
+        const neoSchema = new Neo4jGraphQL({ typeDefs });
+        const schema = await neoSchema.getSchema();
+
+        await expect(
+            neoSchema.assertIndexesAndConstraints({
+                driver,
+                driverConfig: { database: databaseName },
+                options: { create: true },
+            })
+        ).resolves.not.toThrow();
+
+        const session = driver.session({ database: databaseName });
+
+        const cypher = `
+            SHOW INDEXES yield
+                name AS name,
+                type AS type,
+                entityType AS entityType,
+                labelsOrTypes AS labelsOrTypes,
+                properties AS properties
+            WHERE name = "${indexName}"
+            RETURN {
+                 name: name,
+                 type: type,
+                 entityType: entityType,
+                 labelsOrTypes: labelsOrTypes,
+                 properties: properties
+            } as result
+        `;
+
+        try {
+            const result = await session.run(cypher);
+
+            const record = result.records[0].get("result") as {
+                name: string;
+                type: string;
+                entityType: string;
+                labelsOrTypes: string[];
+                properties: string[];
+            };
+
+            expect(record.name).toEqual(indexName);
+            expect(record.type).toBe("FULLTEXT");
+            expect(record.entityType).toBe("NODE");
+            expect(record.labelsOrTypes).toEqual([type.name]);
+            expect(record.properties).toEqual(["id"]);
+
+            await session.run(`
+                CREATE (:${type.name} { id: "${id}" })
+            `);
+        } finally {
+            await session.close();
```

Comment:
I got the feeling that we are testing 2 different things here. The first one is checking for the fulltext index to exists, while the second is an actual fulltext graphql query.
Could it be worth splitting this into 2 separate test cases?

Reply:
I guess it could be, but it follows the pattern of every single other test in this file (I literally copied and pasted), and if I'm being quite honest, I don't have the energy to update every single test! 😆